src/Stan/Pattern/Type.hs

```diff
@@ -89,9 +89,7 @@ infix 5 |::
 -- | 'PatternType' for list @[a]@ or @'String'@.
 listPattern :: PatternType
 listPattern =
-    listNameMeta |:: [ (?) ]
-    |||
-    "String" `baseNameFrom` "GHC.Base" |:: []
+    listNameMeta |:: [ (?) ] ||| stringPattern
```

Comment:
I would keep patterns on separate lines for readability reasons :slightly_smiling_face: 
It makes it easier to see that the list pattern is a choice between two type patterns
```suggestion
    listNameMeta |:: [ (?) ]
    |||
    stringPattern
```